bioimageio/core/_prediction_pipeline.py

```diff
@@ -40,6 +43,52 @@
 )
 
 
+@dataclass(frozen=True)
+class PredictionTensorMetadata:
+    input: Tensor
+    prediction: Tensor
+    input_blocks: List[Block]
+    prediction_blocks: List[Block]
+
+    def __post_init__(self):
+        assert len(self.input_blocks) == len(self.prediction_blocks)
+        assert is_perfect_square(self.total_blocks)
+
+    @property
+    def total_blocks(self):
+        return len(self.input_blocks)
+
+
+@dataclass(frozen=True)
+class PredictionSampleMetadata:
+    input: Sample
+    prediction: Sample
+    input_blocks: List[SampleBlock]
+    prediction_blocks: List[SampleBlock]
+
+    def __post_init__(self):
+        assert len(self.input.members) == len(self.prediction.members)
+        assert len(self.input_blocks) == len(self.prediction_blocks)
+        assert is_perfect_square(self.total_blocks)
+
+    @property
+    def total_blocks(self) -> int:
+        return len(self.input_blocks)
+
+    def get_prediction_tensor_metadata(self, idx: int) -> PredictionTensorMetadata:
```

Comment:
The rest of the code references sample members by their `MemberId`. Would be great to keep that consistent